core/src/test/java/com/statful/client/core/sender/BufferedMetricsSenderAPITest.java

```diff
@@ -60,7 +60,7 @@ public void tearDown() {
     }
 
     @Test
-    public void shouldDiscardIfStandardBufferIsFull() {
```

Comment:
we should, somehow, test the behaviour of trying to empty it when it's empty, more so if we keep the current exception based implementation for when it's empty